File: apps/web/lib/favorite_review.ts
/** 收藏经节复习卡片（§7 P2） */

import { loadFavoriteRefs } from './favorites';

export function favoriteReviewCards(limit = 3): { ref: string; label: string }[] {
  const refs = loadFavoriteRefs();
  if (!refs.length) return [];
  const shuffled = [...refs].sort(() => Math.random() - 0.5);
  return shuffled.slice(0, limit).map((ref) => ({
    ref,
    label: ref.replace(/\./g, ' '),
  }));
}

File: apps/mobile/lib/features/notes/favorite_review.dart
/// 收藏经节复习卡片（对齐 Web favorite_review.ts）。
library;

import 'dart:math';

import '../../core/database/app_database.dart';
import '../../core/ref_label.dart';

List<({String ref, String label})> favoriteReviewCards(
  List<Bookmark> bookmarks, {
  int limit = 3,
}) {
  if (bookmarks.isEmpty) return [];
  final refs = bookmarks.map((b) => b.ref).toList();
  final shuffled = [...refs]..shuffle(Random());
  return shuffled
      .take(limit)
      .map((ref) => (ref: ref, label: formatGroupRefLabel(ref)))
      .toList();
}
